conformance/lang/expressions/binary-bitwise-expression/binary_bitwise_and_expr.balt

```diff
@@ -0,0 +1,1338 @@
+
+int? a = 43;
+
+byte? b = 5;
+
+int:Signed32? c = -21474836;
+int:Signed16? d = -454;
+int:Signed8? e = 100;
+
+int:Unsigned32? f = 21474836;
+int:Unsigned16? g = 54522;
+int:Unsigned8? h = 255;
+
+int? result1 = a & b;
+int? result2 = a & c;
+int? result3 = a & d;
+int? result4 = a & e;
+int? result5 = a & f;
+int? result6 = a & g;
+int? result7 = a & h;
+
+int? result8 = b & b;
+int? result9 = b & c;
+int? result10 = b & d;
+int? result11 = b & e;
+int? result12 = b & f;
+int? result13 = b & g;
+int? result14 = b & h;
+
+int? result15 = c & c;
+int? result16 = c & d;
+int? result17 = c & e;
+int? result18 = c & f;
+int? result19 = c & g;
+int? result20 = c & h;
+
+int? result21 = d & d;
+int? result22 = d & e;
+int? result23 = d & f;
+int? result24 = d & g;
+int? result25 = d & h;
+
+int? result26 = e & e;
+int? result27 = e & f;
+int? result28 = e & g;
+int? result29 = e & h;
+
+int? result30 = f & f;
+int? result31 = f & g;
+int? result32 = f & h;
+
+int? result33 = g & g;
+int? result34 = g & h;
+
+int? result35 = h & h;
+
+function init() {
+    printResult(result1); // @output 1
+    printResult(result2); // @output 40
+    printResult(result3); // @output 42
+    printResult(result4); // @output 32
+    printResult(result5); // @output 0
+    printResult(result6); // @output 42
+    printResult(result7); // @output 43
+
+    printResult(result8); // @output 5
+    printResult(result9); // @output 4
+    printResult(result10); // @output 0
+    printResult(result11); // @output 4
+    printResult(result12); // @output 4
+    printResult(result13); // @output 0
+    printResult(result14); // @output 5
+
+    printResult(result15); // @output -21474836
+    printResult(result16); // @output -21475288
+    printResult(result17); // @output 100
+    printResult(result18); // @output 4
+    printResult(result19); // @output 20712
+    printResult(result20); // @output 236
+
+    printResult(result21); // @output -454
+    printResult(result22); // @output 32
+    printResult(result23); // @output 21474832
+    printResult(result24); // @output 54330
+    printResult(result25); // @output 58
+
+    printResult(result26); // @output 100
+    printResult(result27); // @output 4
+    printResult(result28); // @output 96
+    printResult(result29); // @output 100
+
+    printResult(result30); // @output 21474836
+    printResult(result31); // @output 33808
+    printResult(result32); // @output 20
+
+    printResult(result33); // @output 54522
+    printResult(result34); // @output 250
+
+    printResult(result35); // @output 255
+}
+
+function printResult(int? result) {
+    io:println(result);
+}
+
+Test-Case: output
+Description: Test bitwise and expression where the operands are user-defined type T where the T is a subtype of int.
+Labels: binary-bitwise-expr, byte, int, module-type-defn, int:Signed8, int:Signed32, int:Unsigned16, 
+        int:Unsigned32, singleton-type, unary-minus, union-type
```

Comment:
Re: singletons and user-defined types, shouldn't the following work?

```ballerina
type X 1|2;
type Y 256|3000;

public function main() {
    X a = 1;
    Y b = 256;

    int:Unsigned8 _ = a & b;
}
```

Doesn't work atm. We don't have such tests, do we?

Reply:
Thanx for mentioning it. We don't have such tests. will create an issue and add tests

Reply:
Tests were added and created issue #35458